evil-common.el

```diff
@@ -1927,11 +1927,11 @@ otherwise, it stays behind."
           (setq alist (default-value 'evil-markers-alist)
                 marker (make-marker))
           (evil-add-to-alist 'alist char marker)
-          (setq-default evil-markers-alist alist))
+          (setq-default evil-markers-alist alist)
+          (add-hook 'kill-buffer-hook #'evil-swap-out-markers nil t))
          (t
           (setq marker (make-marker))
           (evil-add-to-alist 'evil-markers-alist char marker))))
-      (add-hook 'kill-buffer-hook #'evil-swap-out-markers nil t)
```

Comment:
We move this hook since the only time we need it is when we have set a global mark